src/main/java/soconnect/ui/PersonCard.java

```diff
@@ -58,73 +59,92 @@ public PersonCard(Person person, int displayedIndex, Logic logic) {
         setAttributes();
     }
 
-    @Override
-    public boolean equals(Object other) {
-        // short circuit if same object
-        if (other == this) {
-            return true;
-        }
-
-        // instanceof handles nulls
-        if (!(other instanceof PersonCard)) {
-            return false;
-        }
-
-        // state check
-        PersonCard card = (PersonCard) other;
-        return id.getText().equals(card.id.getText())
-                && person.equals(card.person);
-    }
-
     /**
      * Sets the attributes in the order required.
      */
     private void setAttributes() {
         int[] order = filterAttributes(orderAttributes());
-        FlowPane[] flowpanes = new FlowPane[5];
+        FlowPane[] flowpanes = new FlowPane[NUMBER_OF_CUSTOMISABLE_ATTRIBUTES + 1];
```

Comment:
I do not believe so. This is referring to the number of customisable attributes and 1 more element for the `FlowPane` that points to none of the four attribute's `FXML` variables. Since this is the only place that will ever need to use this and that there's no intuitive and useful name for it, I would prefer not to extract this out to another constant. `NUMBER_OF_CUSTOMISABLE_ATTRIBUTES` should be sufficient abstraction.

Reply:
Oh. ok. Then that's fine.